gdsfactory/component.py

```diff
@@ -52,7 +52,10 @@ def size(region: kdb.Region, offset: float, dbu=1e3) -> kdb.Region:
 
 
 def boolean_or(region1: kdb.Region, region2: kdb.Region) -> kdb.Region:
-    return region1.__or__(region2)
+    if kdb.Region.interacting(region1, region2):
+        return kdb.Region.merge(region1.__or__(region2))
+    else:
+        return region1.__or__(region2)
```

Comment:
```suggestion
    return (region1.__or__(region2)).merge()
```

`kdb.Region.merge(<other_region>)` is wrong syntax at least according to the docs. It's possible that C++ implicit conversion works, but I would suggest not relying on undocumented behavior

https://klayout.de/doc/code/class_Region.html#method130